pallets/pallet-bank/src/lib.rs

```diff
@@ -568,7 +646,12 @@ impl<T: Config> Pallet<T> {
                     Error::<T>::UnauthorisedUnlock
                 );
                 let unlocked_amount = account_data.locked[index].amount;
-                account_data.free += unlocked_amount;
+
+                if account_data.locked[index].reason == LockReason::Stake {
+                    account_data.reserved += unlocked_amount;
```

Comment:
Always use saturating arithmetic unless we are 100% sure thing won't overflow/underflow